src/common/transformations/src/transformations/common_optimizations/transpose_sinking_pad.cpp

```diff
@@ -0,0 +1,95 @@
+#include "transformations/common_optimizations/transpose_sinking_pad.hpp"
+
+#include <openvino/pass/pattern/op/or.hpp>
+
+#include "itt.hpp"
+#include "openvino/op/util/op_types.hpp"
+#include "openvino/opsets/opset10.hpp"
+#include "openvino/pass/pattern/op/wrap_type.hpp"
+#include "openvino/util/common_util.hpp"
+#include "transformations/common_optimizations/transpose_sinking_utils.hpp"
+#include "transformations/rt_info/transpose_sinking_attr.hpp"
+
+using namespace ov::pass::pattern;
+using namespace ov;
+using namespace ov::opset10;
+using namespace transpose_sinking;
+
+ov::pass::TransposeSinkingPadForward::TransposeSinkingPadForward() {
+    MATCHER_SCOPE(TransposeSinkingPadForward);
+
+    auto main_node_label = wrap_type<Pad>(IfFirstNodeInputHasTranspose);
```

Comment:
Maybe here is just simpler to create main_node_label = ({transpose, any, any}) ? Function that searches for the input Transposes is useful if we have multiple inputs and we don't know what input is Transpose

Reply:
agree